### PR TITLE
fix: NetCDFIO

### PR DESCRIPTION
**Pull request by haakon-e**
Fix a bug where `namelist` and `paramlist` files were not found by `NetCDFIO.__init__` if they were not in the same directory as the SCAMPy base directory.

Also made some stylistic changes, including using `pathlib` instead of `os` for directory operations.

### CODE REVIEW

File: Cases.pxd [hunk 1]
```diff
@@ -41,6 +41,18 @@ cdef class Bomex(CasesBase):
...
+    cpdef initialize_reference(self, Grid Gr, ReferenceState Ref, NetCDFIO_Stats Stats) 
+    cpdef initialize_profiles(self, Grid Gr, GridMeanVariables GMV, ReferenceState Ref )
+    cpdef initialize_surface(self, Grid Gr,  ReferenceState Ref )
+    cpdef initialize_forcing(self, Grid Gr,  ReferenceState Ref, GridMeanVariables GMV )
+    cpdef initialize_io(self, NetCDFIO_Stats Stats)
+    cpdef io(self, NetCDFIO_Stats Stats)
+    cpdef update_surface(self, GridMeanVariables GMV, TimeStepping TS)
+    cpdef update_forcing(self, GridMeanVariables GMV,  TimeStepping TS)
+##
+
```
Comment (haakon-e) [inline Cases.pxd:55]:
```suggestion
```

File: Cases.pyx [hunk 2]
```diff
@@ -38,6 +38,8 @@ def CasesFactory(namelist, paramlist):
         return SP(paramlist)
     elif namelist['meta']['casename'] == 'DryBubble':
         return DryBubble(paramlist)
+    elif namelist['meta']['casename'] == 'StochasticBomex':  # stochastic case
+        return StochasticBomex(paramlist)
```
Comment (haakon-e) [inline Cases.pyx:42]:
```suggestion
```

File: Cases.pyx [hunk 3]
```diff
@@ -390,6 +392,35 @@ cdef class Bomex(CasesBase):
...
+    
+    cpdef io(self, NetCDFIO_Stats Stats):
+        super(StochasticBomex, self).io(Stats)
+    
+    cpdef update_surface(self, GridMeanVariables GMV, TimeStepping TS):
+        super(StochasticBomex, self).update_surface(GMV, TS)
+    
+    cpdef update_forcing(self, GridMeanVariables GMV, TimeStepping TS):
+        super(StochasticBomex, self).update_forcing(GMV, TS)
+
```
Comment (haakon-e) [inline Cases.pyx:423]:
```suggestion
```

File: Turbulence_PrognosticTKE.pxd [hunk 4]
```diff
@@ -62,6 +62,10 @@ cdef class EDMF_PrognosticTKE(ParameterizationBase):
         double dt_upd
         double constant_plume_spacing
         double aspect_ratio
+
+        # stochastic parameters
+        double stochastic_noise
+
```
Comment (haakon-e) [inline Turbulence_PrognosticTKE.pxd:68]:
```suggestion
```

File: Turbulence_PrognosticTKE.pyx [hunk 5]
```diff
@@ -167,6 +167,10 @@ cdef class EDMF_PrognosticTKE(ParameterizationBase):
         self.pressure_buoy_coeff = paramlist['turbulence']['EDMF_PrognosticTKE']['pressure_buoy_coeff']
         self.aspect_ratio = paramlist['turbulence']['EDMF_PrognosticTKE']['aspect_ratio']
 
+        ## Haakon: Stochastic closure
+        self.stochastic_noise = paramlist['turbulence']['EDMF_PrognosticTKE']['stochastic_noise']
+        ##
+
```
Comment (haakon-e) [inline Turbulence_PrognosticTKE.pyx:173]:
```suggestion
```

File: Turbulence_PrognosticTKE.pyx [hunk 6]
```diff
@@ -1301,6 +1305,9 @@ cdef class EDMF_PrognosticTKE(ParameterizationBase):
         return
 
     cpdef compute_entrainment_detrainment(self, GridMeanVariables GMV, CasesBase Case):
+        # Compute entrainment and detrainment parameters.
+        #
+        # Here, you can inject a stochastic closure.
```
Comment (haakon-e) [inline Turbulence_PrognosticTKE.pyx:1310]:
```suggestion
```

File: Turbulence_PrognosticTKE.pyx [hunk 7]
```diff
@@ -1381,6 +1388,10 @@ cdef class EDMF_PrognosticTKE(ParameterizationBase):
                     ret = self.entr_detr_fp(input)
                     self.entr_sc[i,k] = ret.entr_sc
                     self.detr_sc[i,k] = ret.detr_sc
+                    ## Haakon (stochastic closure)
+                    self.entr_sc[i,k] *= np.random.lognormal(1.0, self.stochastic_noise)
+                    self.detr_sc[i,k] *= np.random.lognormal(1.0, self.stochastic_noise)
+                    ## End: Haakon
```
Comment (haakon-e) [inline Turbulence_PrognosticTKE.pyx:1394]:
```suggestion
```

File: generate_namelist.py [hunk 8]
```diff
@@ -63,6 +63,8 @@ def main():
 
     if case_name == 'Bomex':
         namelist = Bomex(namelist_defaults)
+    elif case_name == 'StochasticBomex':
+        namelist = StochasticBomex(namelist_defaults)
```
Comment (haakon-e) [inline generate_namelist.py:67]:
```suggestion
```

File: generate_namelist.py [hunk 9]
```diff
@@ -137,6 +139,16 @@ def Bomex(namelist_defaults):
...
+
+def StochasticBomex(namelist_defaults):
+    # Haakon: stochastic closure
+    namelist = Bomex(namelist_defaults)
+    namelist['meta']['simname'] = 'StochasticBomex'
+    namelist['meta']['casename'] = 'StochasticBomex'
+
+    return namelist
+
+
```
Comment (haakon-e) [inline generate_namelist.py:151]:
```suggestion
```

File: generate_paramlist.py [hunk 10]
```diff
@@ -91,12 +91,23 @@ def main():
         paramlist = SP(paramlist_defaults)
     elif case_name == 'DryBubble':
         paramlist = DryBubble(paramlist_defaults)
+    elif case_name == 'StochasticBomex':  # Haakon: Stochastic closure
+        paramlist = StochasticBomex(paramlist_defaults)
```
Comment (haakon-e) [inline generate_paramlist.py:95]:
```suggestion
```

File: generate_paramlist.py [hunk 11]
```diff
@@ -91,12 +91,23 @@ def main():
...
 
+
+def StochasticBomex(paramlist_defaults):
+    # Haakon: Stochastic closure
+    paramlist = Bomex(paramlist_defaults)
+    paramlist["meta"]["casename"] = "StochasticBomex"
+    paramlist["turbulence"]["EDMF_PrognosticTKE"]["stochastic_noise"] = 1.0
+    return paramlist
+
+
```
Comment (haakon-e) [inline generate_paramlist.py:110]:
```suggestion
```

**Review by haakon-e — COMMENTED** [review]
remove these

**Comment by trontrytel** [comment]
Looks good to me. Could you also update the calls to `main1d` in tests? (Talking about all the `test_plot_...` semi automatic scripts inside the `tests/plots` folder and the tiny bit of unit tests that we have in `tests/unit` folder).  You can run all the tests by `py.test -s -v` inside the `tests` folder. Individual test can be run by `py.test -s -v plots/test_plot_TRMM_LBA.py` inside the `tests` folder.

I think there may be some inconsistency in prints syntax from python2 to python3. If we all agree I can just change it all to python3. I have been running scampy on python3 for a long time anyway.

**Review by trontrytel — CHANGES_REQUESTED** [review]
Aside from fixing the calls inside the tests, everything looks good

**Review by yairchn — APPROVED** [review]
LGTM

**Pull request merged**